packages/browser-pool/src/browser-pool.ts

```diff
@@ -694,12 +705,21 @@ export class BrowserPool<
         return this.browserPlugins[pluginIndex];
     }
 
-    private _pickBrowserWithFreeCapacity(browserPlugin: BrowserPlugin, options?: { proxyTier?: number }) {
+    private _pickBrowserWithFreeCapacity(browserPlugin: BrowserPlugin, options?:
+        { proxyTier?: number; proxyUrl?: never } | { proxyTier?: never; proxyUrl?: string | null }) {
```

Comment:
not that it would matter much, but i dont like those line breaks very much. if we want to break lines on the method signature, it should be consistent, something like this:

```suggestion
    private _pickBrowserWithFreeCapacity(
        browserPlugin: BrowserPlugin, 
        options?: { proxyTier?: number; proxyUrl?: never } | { proxyTier?: never; proxyUrl?: string | null },
    ) {
```